tests/func/test_repo.py

```diff
@@ -1,71 +1,60 @@
-from dvc.scm.tree import WorkingTree
-from dvc.stage import Stage
-from tests.basic_env import TestDvcGit
-
-
-class TestCollect(TestDvcGit):
-    def setUp(self):
-        super(TestCollect, self).setUp()
-        self.dvc.add(self.FOO)
-        self.dvc.run(
-            deps=[self.FOO],
-            outs=[self.BAR],
-            cmd="python code.py {} {}".format(self.FOO, self.BAR),
-        )
-        self.dvc.scm.add([".gitignore", self.FOO + ".dvc", self.BAR + ".dvc"])
-        self.dvc.scm.commit("foo.dvc and bar.dvc")
-        self.dvc.scm.checkout("new_branch", True)
-        self.dvc.run(
-            deps=[self.BAR],
-            outs=["buzz"],
-            cmd="python code.py {} {}".format(self.BAR, "buzz"),
-        )
-        self.dvc.scm.add([".gitignore", "buzz.dvc"])
-        self.dvc.scm.commit("add buzz")
-        self.dvc.scm.checkout("master")
-
-    def _check(self, branch, target, with_deps, expected):
-        if branch:
-            self.dvc.tree = GitTree(self.dvc.scm.repo, branch)
-        else:
-            self.dvc.tree = WorkingTree()
-        result = self.dvc.collect(target + ".dvc", with_deps=with_deps)
-        self.assertEqual([[str(j) for j in i.outs] for i in result], expected)
-        return result
-
-    def test(self):
-        self._check("", self.BAR, True, [[self.FOO], [self.BAR]])
-        self._check("master", self.BAR, True, [[self.FOO], [self.BAR]])
-        self._check(
-            "new_branch", "buzz", True, [[self.FOO], [self.BAR], ["buzz"]]
+from dvc.system import System
+from dvc.utils.compat import fspath
+
+
+def test_destroy(tmp_dir, dvc):
+    dvc.config.set("cache", "type", "symlink")
+
+    tmp_dir.dvc_gen("file", "text")
+    tmp_dir.dvc_gen({"dir": {"file": "lorem", "subdir/file": "ipsum"}})
+
+    dvc.destroy()
+
+    # Remove all the files related to DVC
+    for path in [".dvc", "file.dvc", "dir.dvc"]:
+        assert not (tmp_dir / path).exists()
+
+    # Leave the rest of the files
+    for path in ["file", "dir/file", "dir/subdir/file"]:
+        assert (tmp_dir / path).is_file()
+
+    # Make sure that data was unprotected after `destroy`
+    for path in ["file", "dir", "dir/file", "dir/subdir", "dir/subdir/file"]:
+        assert not System.is_symlink(fspath(tmp_dir / path))
+
+
+def test_collect(tmp_dir, scm, dvc, run_copy):
+    def collect_outs(*args, **kwargs):
+        return set(
+            str(out.path_info)
+            for stage in dvc.collect(*args, **kwargs)
+            for out in stage.outs
         )
-        result = self._check("new_branch", "buzz", False, [["buzz"]])
-        self.assertEqual([str(i) for i in result[0].deps], ["bar"])
 
+    tmp_dir.dvc_gen("foo", "foo")
+    run_copy("foo", "bar")
+    assert collect_outs("bar.dvc", with_deps=True) == {"foo", "bar"}
+
+    run_copy("bar", "buzz")
+    assert collect_outs("buzz.dvc", with_deps=True) == {"foo", "bar", "buzz"}
+    assert collect_outs("buzz.dvc", with_deps=False) == {"buzz"}
```

Comment:
The "buzz" was added in a branch originally. And it was a collect with a branch assigned.

Reply:
I think it doesn't make sense to do the branching, but I'll do it just to not be a blocker for this PR